Compare View Revamp: wire up maps and modification popovers

arottersman opened the pull request:
## Overview
- Add map views to each scenario column in the new compare view
- Add modifications popovers to each map

Connects #2071 

### Demo

![screen shot 2017-08-15 at 4 38 54 pm](https://user-images.githubusercontent.com/7633670/29336214-11e29a3a-81db-11e7-8be9-bb1a934740ab.png)

![screen shot 2017-08-15 at 4 40 08 pm](https://user-images.githubusercontent.com/7633670/29336208-0ef5969c-81db-11e7-912d-b3a585203a44.png)

### Notes

@jfrankl, two things:
-  If there's no modifications on the scenario, I opted not to show any popover. Let me know if you'd like to see one that says "No modifications" or something.
- In the wireframe, you've dropped the mod type label "Land Cover" or "Conservation Practices". Was this intentional?
<img width="331" alt="screen shot 2017-08-15 at 5 17 50 pm" src="https://user-images.githubusercontent.com/7633670/29336993-c49786f2-81dd-11e7-9622-0bc9b4e09daa.png">
I kept them in for now.



## Testing Instructions
* Pull and `./scripts/bundle.sh`
 * In the compare view for a TR-55 project with some modified scenarios...
* Confirm the mini-maps' extents are the modifications, if they exist.
* Hover over a map with mods (or tap on mobile) and confirm a popover appears with the areas for each type

kellyi commented on src/mmw/js/src/compare/templates/compareModificationsPopover.html:
```diff
@@ -0,0 +1,23 @@
+{% macro modificationsTable(caption, modifications) %}
```
Nice! Didn't know this was possible.

arottersman replied:
Same! I stole this from the original implementation of the mod table. Nunjunks has a quite a few tricks up its sleeves

kellyi commented:
This is working really well! Here's a project with a scenario including all modifications types:

![screen shot 2017-08-16 at 10 10 21 am](https://user-images.githubusercontent.com/4165523/29367537-34cbbdc0-826b-11e7-8e34-decef94cfe35.png)

kellyi commented:
Here's one with a modification larger than the initial AOI; map extent's changed correctly:

![screen shot 2017-08-16 at 10 18 03 am](https://user-images.githubusercontent.com/4165523/29367878-40de39d4-826c-11e7-89e9-59a43304ce99.png)

kellyi commented:
I think this is due to something in the compare view's CSS rules but on Safari the font weight for the precipitation slider and some of the other text seems to drop slightly yet notice-ably when one of the popovers is open:

<img width="1030" alt="screen shot 2017-08-16 at 10 20 42 am" src="https://user-images.githubusercontent.com/4165523/29368083-ba4fc5f8-826c-11e7-8ff9-931023327c59.png">

<img width="1043" alt="screen shot 2017-08-16 at 10 20 54 am" src="https://user-images.githubusercontent.com/4165523/29368092-bdc274c4-826c-11e7-8600-633a8e4032b1.png">

arottersman commented:
Thanks, @kellyi! I went over this with Jeff. I'll be adding a commit that adds some placeholder text to popovers for  scenarios that can't/don't have modifications with the intention that someday:
- 100% Forest will have some description of itself
- Current Conditions will have some description of itself
- Regular scenarios without modifications will say No modifications + something like "to add modifications...."

I'll also look into that safari styling quirk.

arottersman commented on src/mmw/js/src/compare/views.js:
```diff
@@ -170,6 +172,31 @@ var CompareModificationsPopoverView = Marionette.ItemView.extend({
+        var title = this.model.get('name'),
+            text;
+
+        if (this.model.get('is_pre_columbian')) {
```
This could be easily done in the template as well, but I figured it was easier to debug if we kept the logic here.

arottersman replied:
Actually on second thought I'm going to move it to the template.

arottersman commented:
- Added a commit to fix the safari quirk
- Added additional popovers
![screen shot 2017-08-16 at 1 50 36 pm](https://user-images.githubusercontent.com/7633670/29378511-58e53f0c-828d-11e7-989a-9a91090a4f1c.png)
![screen shot 2017-08-16 at 1 50 46 pm](https://user-images.githubusercontent.com/7633670/29378510-58e32c12-828d-11e7-8a3b-5a32ec35af8a.png)
![screen shot 2017-08-16 at 1 50 31 pm](https://user-images.githubusercontent.com/7633670/29378534-62d85904-828d-11e7-9e24-7fef9fe4408e.png)

arottersman commented:
@kellyi, before I merge, mind giving this a final look with the new popovers?

kellyi commented:
Sure, happy to test it out again!

kellyi commented:
+1, working well! Here's the 100% forest cover popover on IE11:

<img width="1053" alt="screen shot 2017-08-21 at 11 39 44 am" src="https://user-images.githubusercontent.com/4165523/29527085-ad2ca524-8665-11e7-9873-77528896589a.png">

Also working in FF, Safari, and Chrome.

arottersman commented:
Thanks for all the testing and review! Squashed the fixup. Will merge this once the build completes.
